utils/databake/src/primitives.rs

```diff
@@ -55,6 +64,15 @@ where
     T: Bake,
 {
     fn bake(&self, ctx: &CrateEnv) -> TokenStream {
+        if core::mem::size_of::<T>() == core::mem::size_of::<u8>()
+            && core::any::type_name::<T>() == core::any::type_name::<u8>()
```

Comment:
Thought: I was hoping we could do this safely by going through `dyn Any`, but it seems `Any` is not `?Sized`, so it won't work on a `[u8]`. Can't find out why not. And of course it won't work on `&[u8]` due to the non-static lifetime.

Reply:
dyn Any for ?Sized would need multiple metadata bytes, which is something folks want but doesn't exist right now